cpp/include/raft/sparse/csr.hpp

```diff
@@ -178,5 +178,141 @@ void weak_cc(Index_* labels,
     labels, row_ind, row_ind_ptr, nnz, N, 0, N, stream, [](Index_) { return true; });
 }
 
+/**
+ * @brief CSR representation of a MFG.
+ *
+ * @tparam IdxT node index type
+ *
+ * Same as `mfg_ellpack` except the neighbors are in CSR format.
+ *
+ * @note Valid IDs: [0, 2^(sizeof(IdxT) * 8 - 1) - 2] (see `INVALID_ID`)
+ * @note This object does NOT own any of the underlying pointers and thus is
+ *       left to the calling code for their lifetime management.
+ * @note This is always an in-graph (`offsets` over the destination nodes).
+ *       See `fg_csr` for further explanation on this.
```

Comment:
What is `fg_csr`? `mfg_csr`? I think we can combine `mfg_csr` and `mfg_csr_rev` as something like `mfg_adjacency_list` with a template parameter similar to cugraph's graph_view_t.

Reply:
This makes sense to me- and perhaps have them extend the csr implementation after it's moved to RAFT. I plan to create a PR for new (d)csr/coo view structures this week.